source/postprocess/dynamic_topography.cc

```diff
@@ -79,7 +116,130 @@ namespace aspect
+                      epsilon_phi_u[k] = fe_volume_values[this->introspection().extractors.velocities].symmetric_gradient(k,q);
+                      div_phi_u[k] = fe_volume_values[this->introspection().extractors.velocities].divergence (k, q);
+                    }
+
+                  for (unsigned int i = 0; i<dofs_per_cell; ++i)
+                    {
+                      //Viscous stress part
+                      local_vector(i) += 2.0 * eta * ( epsilon_phi_u[i] * in_volume.strain_rate[q]
+                                                       - (is_compressible ? 1./3. * div_phi_u[i] * div_solution[q] : 0.0) ) * fe_volume_values.JxW(q);
+                      //Pressure and compressibility parts
+                      local_vector(i) -= div_phi_u[i] * in_volume.pressure[q] * fe_volume_values.JxW(q);
+                      //Force part
+                      local_vector(i) -= density * gravity * phi_u[i] * fe_volume_values.JxW(q);
+                    }
+                }
+              // Assemble the mass matrix for cell face. Since we are using GLL
+              // quadrature, the mass matrix will be diagonal, and we can just assemble it into a vector.
+              for (unsigned int q=0; q < n_face_q_points; ++q)
+                for (unsigned int i=0; i<dofs_per_cell; ++i)
+                  local_mass_matrix(i) += fe_face_values[this->introspection().extractors.velocities].value(i,q) *
+                                          fe_face_values[this->introspection().extractors.velocities].value(i,q) *
+                                          fe_face_values.JxW(q);
+
+              cell->distribute_local_to_global( local_vector, rhs_vector );
+              cell->distribute_local_to_global( local_mass_matrix, mass_matrix );
+            }
+
+      rhs_vector.compress(VectorOperation::add);
+      mass_matrix.compress(VectorOperation::add);
 
+      // Since the mass matrix is diagonal, we can just solve for the stress vector by dividing.
+      const IndexSet local_elements = mass_matrix.locally_owned_elements();
+      for (unsigned int k=0; k<local_elements.n_elements(); ++k)
+        {
+          const unsigned int global_index = local_elements.nth_index_in_set(k);
+          if ( mass_matrix[global_index] > 1.e-15)
+            distributed_topo_vector[global_index] = rhs_vector[global_index]/mass_matrix[global_index];
+        }
+      distributed_topo_vector.compress(VectorOperation::insert);
+      topo_vector = distributed_topo_vector;
+
+      // Now loop over the cells again and solve for the dynamic topography.
+      // We solve for it on the support points of the system, since it can be
+      // directly put into a system vector of the right size.
+      std::vector< Point<dim-1> > face_support_points = this->get_fe().base_element( this->introspection().base_elements.temperature ).get_unit_face_support_points();
```

Comment:
There is no problem as long as they are interpolatable onto each other. I have verified that it works fine with linear temperature elements.